Lib/urllib/parse.py

```diff
@@ -155,9 +155,19 @@ def password(self):
     def hostname(self):
         hostname = self._hostinfo[0]
         if not hostname:
-            hostname = None
-        elif hostname is not None:
+            return None
+        # Scoped IPv6 address may have zone info, which must not be lowercased
+        # like http://[fe80::822a:a8ff:fe49:470c%tESt]:1234/keys
+        if isinstance(hostname, str):
+            (hostname, *zone) = hostname.split('%', 1)
```

Comment:
Drop parentheses at left of `=`, the module doesn't use this style.

Reply:
I recommend considering `hostname.partition('%')` -- it makes code cleaner.

Reply:
fixed